lib/handler/decorator-handler.js

```diff
@@ -32,7 +127,7 @@ module.exports = class DecoratorHandler {
 
   onResponseStarted (...args) {
     assert(!this.#onCompleteCalled)
-    assert(!this.#onErrorCalled)
+    // assert(!this.#onErrorCalled)
```

Comment:
why?

Reply:
We have a bug in the h2 client that wasn't caught before.

Reply:
@metcoder95: client-h2. calls onResponseStarted after onError

Reply:
Hmm, interesting; I imagine this happens if something failed before dispatching the request, or how do you find it?

Reply:
I applied the assertions to the core request handling:

https://github.com/nodejs/undici/pull/3636/files#diff-ea7696549c3a0b60a4a7e07cc79b6d4e950c7cb1068d47e368a510967d77e7e5R49-R51